File: 2022/09/04/islandPerimeter.js
const islandPerimeter = function(grid) {
    let result = 0;

    for (let i = 0; i < grid.length; i++) {
        for (let j = 0; j < grid[0].length; j++) {
            if (grid[i][j]) {
                if (i === 0 || grid[i - 1][j] === 0) {
                    result++;
                }

                if (j === 0 || grid[i][j - 1] === 0) {
                    result++;
                }

                if (i === grid.length - 1 || grid[i + 1][j] === 0) {
                    result++;
                }

                if (j === grid[0].length - 1 || grid[i][j + 1] === 0) {
                    result++;
                }
            }
        }
    }

    return result;
};

let result = islandPerimeter([[0,1,0,0],[1,1,1,0],[0,1,0,0],[1,1,0,0]]);
console.assert(result === 16);

result = islandPerimeter([[1]]);
console.assert(result === 4);

result = islandPerimeter([[1, 0]]);
console.assert(result === 4);

File: 2023/06/29/go/island_perimeter.go
/**
 * https://leetcode.com/problems/island-perimeter/
 * Author: Andrew Jarombek
 * Date: 6/29/2023
 */

package _go

func islandPerimeter(grid [][]int) int {
	result := 0

	for i, row := range grid {
		for j, value := range row {
			if value == 1 {
				if i == 0 || grid[i-1][j] == 0 {
					result++
				}

				if j == 0 || grid[i][j-1] == 0 {
					result++
				}

				if i == len(grid)-1 || grid[i+1][j] == 0 {
					result++
				}

				if j == len(row)-1 || grid[i][j+1] == 0 {
					result++
				}
			}
		}
	}

	return result
}
